repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java

```diff
@@ -256,17 +253,17 @@ private void personaPolicyToESDslClauses(List<AtlasEntity> policies,
                         if (isHierarchical) {
                             asset = asset.substring(0, asset.length() - 2);
                         }
-                        boolean isWildcard = asset.contains("*") || asset.contains("?");
-                        if (isWildcard) {
-                            clauseList.add(mapOf("wildcard", mapOf(QUALIFIED_NAME, asset)));
-                        } else if (useHierarchicalQualifiedNameFilter) {
+                        boolean isWildcard = (asset.contains("*") && asset.indexOf("*") != asset.length() - 1) || asset.contains("?");
+                        if (isHierarchical) {
                             terms.metadataPolicyQualifiedNames.add(asset);
+                        } else if (isWildcard) {
+                            clauseList.add(mapOf("wildcard", mapOf(QUALIFIED_NAME, asset)));
                         } else {
                             terms.qualifiedNames.add(asset);
                         }
 
-                        if (!useHierarchicalQualifiedNameFilter || isWildcard) {
-                            clauseList.add(mapOf("wildcard", mapOf(QUALIFIED_NAME, asset + "/*")));
+                        if (isWildcard) {
+                            clauseList.add(mapOf("term", mapOf(QUALIFIED_NAME_HIERARCHY_PROPERTY_KEY, asset)));
```

Comment:
As per check for wildcard, it will be true if `*` or `?` is present in the asset, so `term` clause will not be sufficient to match here. This will be not be useful to capture children of the matched asset with `term` clause.